### PR TITLE
Add CompDbRules as outputs of the generator

### PR DESCRIPTION
**Pull request by Javier-varez**
So that dbt can create a more fine-grained compilation database that does not confuse IDE's and tools like reqtraq

### CODE REVIEW

File: RULES/core/context.go [hunk 1]
```diff
@@ -453,6 +461,15 @@ func (ctx *context) ninjaFile() string {
 	return ninjaFile.String()
 }
 
+func (ctx *context) RegisterCompDbRule(rule *BuildRule) {
+	ctx.compDbBuildRules[rule.Name] = rule
+}
+
+func (ctx *context) GetCompDbRule(name string) (*BuildRule, bool) {
+	buildRule, ok := ctx.compDbBuildRules[name]
```
Comment (jfrohnhofen) [inline RULES/core/context.go:469]: just return `ctx.compDbBuildRules[name]`

File: RULES/core/main.go [hunk 2]
```diff
@@ -137,6 +138,11 @@ func GeneratorMain(vars map[string]interface{}) {
 			}
 		}
 		output.NinjaFile = ctx.ninjaFile()
+
+		output.CompDbRules = []string{}
+		for name, _ := range ctx.compDbBuildRules {
```
Comment (jfrohnhofen) [inline RULES/core/main.go:143]: rm the ", _" before ":= range"

**Review by jfrohnhofen — APPROVED** [review]

File: RULES/core/context.go (hunk 1 above)
Comment (Javier-varez) [reply RULES/core/context.go:469]: I also thought that would work, but turns out that invokes the other method of a map that aborts and then I get:
```
dbt-rules/RULES/core/context.go:469:9: not enough return values
        have (*BuildRule)
        want (*BuildRule, bool)
```

**Comment by jfrohnhofen** [comment]
Oh man... nvm then

**Pull request merged**